app/services/market_analysis_core.py

```
@@ -37,36 +37,73 @@
+        exchange: str, 
+        endpoint: str, 
+        params: Optional[Dict] = None
+    ) -> Dict[str, Any]:
+        """Fetch data from specific exchange with rate limiting."""
+        config = self.exchange_configs[exchange]
+        url = config["base_url"] + endpoint
+        
+        async with aiohttp.ClientSession() as session:
+            async with session.get(url, params=params, timeout=aiohttp.ClientTimeout(total=10)) as response:
+                if response.status != 200:
+                    raise Exception(f"{exchange} API error: {response.status}")
```

Ruff (0.12.2)

Create your own exception

(TRY002)